versioned/mongodb/src/main/java/com/dremio/nessie/versioned/store/mongodb/MongoDBStore.java

```diff
@@ -261,21 +265,34 @@ public void load(LoadStep loadstep) throws ReferenceNotFoundException {
     Preconditions.checkArgument(type.getObjectClass().isAssignableFrom(value.getClass()),
         "ValueType %s doesn't extend expected type %s.", value.getClass().getName(), type.getObjectClass().getName());
 
-    // TODO: Handle ConditionExpressions.
+    final MongoCollection<V> collection = getCollection(type);
     if (conditionUnAliased.isPresent()) {
-      throw new UnsupportedOperationException("ConditionExpressions are not supported with MongoDB yet.");
+      MongoDBAliasCollectorImpl collector = new MongoDBAliasCollectorImpl();
+      ConditionExpressionAliasVisitor conditionExpressionAliasVisitor = new MongoDBConditionExpressionAliasVisitor();
+      ConditionExpression aliased = conditionUnAliased.get().accept(conditionExpressionAliasVisitor, collector);
+      BsonConditionExpressionVisitor bsonConditionExpressionVisitor = new BsonConditionExpressionVisitor();
+      await(collection.replaceOne(aliased.accept(bsonConditionExpressionVisitor), value, new ReplaceOptions().upsert(true)));
+    } else {
+      // Use upsert so that if an item does not exist, it will be insert.
+      await(collection.replaceOne(Filters.eq(Store.KEY_NAME, ((HasId) value).getId()), value, new ReplaceOptions().upsert(true)));
```

Comment:
Can you collapse these two cases into one? Currently your first block ignores the ID equality filter which is implicit.